fx.go

```diff
@@ -19,3 +19,149 @@
 // THE SOFTWARE.
 
 package fx // import "go.uber.org/fx"
+
+import (
+	"io"
+	"reflect"
+
+	"github.com/pkg/errors"
+	"github.com/uber-go/tally"
+
+	"go.uber.org/dig"
+	"go.uber.org/fx/config"
+	"go.uber.org/fx/metrics"
+	"go.uber.org/fx/ulog"
+	"go.uber.org/zap"
+)
+
+// Component is a function that initializes and returns objects to be shared across the code
+type Component interface{}
+
+// Module is a building block or primary function performed by a service built on fx
+type Module interface {
+	Name() string
+	Constructor() []Component
```

Comment:
This can be easily achieved with an option: `WithConfigurableModule` that takes a module ctor with a config. It is true, that our control flow is a bit confusing right now: we make user to create modules for our service, instead of e.g. giving us a module ctor.